app/controllers/analytics_controller.rb

```diff
@@ -0,0 +1,24 @@
+class AnalyticsController < ApplicationController
+  before_action :custom_method, :only => [:show, :index]
+
+  def show
+  end
+
+  def index
+    @last_days = 30
+    @full_width_content = true
+    @visits = Visit.where(started_at: @last_days.days.ago..Time.now)
+    @views = Ahoy::Event.where(name: '$view' )
+  end
+
+private
```

Comment:
Why is this called "custom_method"? Seems like a generic name for a specific method.

I also think there's a more elegant way to require a user to be an admin. Here's an excerpt from the `config/routes.rb` file, lines 34 - 39
```
mount Split::Dashboard, at: 'split', :anchor => false, :constraints => lambda { |request|
    request.env['warden'].authenticated? # are we authenticated?
    request.env['warden'].authenticate! # authenticate if not already
    # or even check any other condition
    request.env['warden'].user.admin?
  }
```